server/utils.py

```diff
@@ -14,18 +14,25 @@
 CHUNK_PREFIX = 'chunk.'
 
 
-def generate_auth_token(access_code):
+def generate_auth_token(server_id, user, name=None, email=None, duration_days=None):
     auth_token = base64.urlsafe_b64encode(os.urandom(12))
     current_date = dt.datetime.today()
-    expiry_date = current_date + dt.timedelta(days=1)  # Code expires in 24 hours
-    g.db.execute('insert into access (site_access_code,auth_token,date_created,date_expired) '
-                 'values (?,?,?,?)',
-                 (access_code, auth_token, current_date.strftime("%Y-%m-%dT%H:%M:%SZ"),
-                  expiry_date.strftime("%Y-%m-%dT%H:%M:%SZ")))
-    g.db.commit()
+    if not duration_days:
+        duration_days = 1
+    expiry_date = current_date + dt.timedelta(duration_days)  # Code expires in 24 hours by default
+    save_auth_token(server_id, user, name, email, auth_token, current_date, expiry_date)
     return auth_token, expiry_date
 
 
+def save_auth_token(server_id, user, name, email, auth_token, date_created, expiry_date):
+    server_name = app.config['SERVER_TOKENS'][server_id]['name']
+    server_address = app.config['SERVER_TOKENS'][server_id]['address']
```

Comment:
This function should take the `server_token`, and we should look up the `server_id` from the dict and not have an address field.